samples/net/sockets/echo/README.rst

```diff
@@ -0,0 +1,78 @@
+.. _sockets-echo-sample:
+
+Socket Echo Server
+##################
+
+Overview
+********
+
+The sockets/echo sample application for Zephyr implements an IPv4 TCP echo
+server using BSD Sockets like API. The purpose of this sample is to show
+how it's possible to develop sockets application portable to both POSIX
+and Zephyr. As such, it is kept minimal and supports only IPv4 and TCP.
+
+The source code for this sample application can be found at:
+:file:`samples/net/sockets/echo`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+- or, a board with hardware networking
+
+Building and Running
+********************
+
+Build Zephyr version of sockets/echo application like this:
+
+.. code-block:: console
+
+    $ cd $ZEPHYR_BASE/samples/net/sockets/echo
+    $ make pristine
+    $ make BOARD=<board_to_use>
+
+``board_to_use`` defaults to ``qemu_x86``. In this case, you can run the
+application in QEMU using ``make run``. If you used another BOARD, you
+will need to consult its documentation for application deployment
+instructions.
```

Comment:
how about adding a reference to the support boards here:

     You can read about Zephyr support for specific boards in the
     documentation at :ref:`boards`.